app/javascript/app/pages/national-circumstances/population/population-selectors.js

```diff
@@ -37,15 +61,82 @@ const getYearParam = ({ location }) =>
 export const getYearSelected = createSelector(
   [ getYearOptions, getYearParam ],
   (years, year) => {
+    if (!years) return null;
     if (!year) return years[0];
     return years.find(y => y.value === year);
   }
 );
 
+const findYearData = (data, year) =>
+  data.categoryYears.find(d => String(d.year) === year.value);
+const getTotalPopulation = createSelector(
+  [ getTotalPopulationData, getYearSelected ],
+  (data, year) => {
+    if (!data || !year) return null;
+    const yearData = findYearData(data, year);
+    return {
+      description: 'Total South Africa population',
+      value: yearData ? withCommas(yearData.value) : 'No data'
+    };
+  }
+);
+
+const getGrowthRate = createSelector([ getTotalGrowthData, getYearSelected ], (
+  data,
+  year
+) =>
+  {
+    if (!data || !year) return null;
+    const yearData = findYearData(data, year);
+    return {
+      description: 'South Africa population growth rate',
+      value: yearData ? `${withCommas(yearData.value)}%` : 'No data'
+    };
+  });
+
+const getRegionPopulation = createSelector(
+  [ selectNationalCircumstances, getYearSelected ],
+  (data, year) => {
+    if (!data || !year) return null;
+    const populationRegionData = {};
+    data.forEach(d => {
+      const yearValue = d.categoryYears.find(
+        c => String(c.year) === year.value
+      );
+      const totalRegionPopulationData = data.find(
+        p => p.name === 'pop_total' && p.location.name === d.location.name
+      );
+      const totalRegionPopulationyearData = totalRegionPopulationData.categoryYears.find(
```

Comment:
add a guard here: `totalRegionPopulationData && totalRegionPopulationData.categoryYears` as it fails if `totalRegionPopulationData` is undefined